java/vector/src/main/java/org/apache/arrow/vector/schema/VectorLayout.java

```diff
@@ -58,7 +58,7 @@ public static VectorLayout dataVector(int typeBitWidth) {
       case 64:
         return VALUES_64;
       default:
-        throw new IllegalArgumentException("only 8, 16, 32, or 64 bits supported");
+        return new VectorLayout(DATA, typeBitWidth);
```

Comment:
The existing C++ implemetation allows any typeBitWidth, I followed and remove this exception. Is this correct?